lib/piezo.js

```diff
@@ -229,38 +301,39 @@ Piezo.prototype.note = function(note, duration) {
   return this.tone(tone, duration);
 };
 
+function throwNeedsToneFrequencyError() {
+  throw new Error(
+      "Piezo.tone: Controller must define either frequency or tone function."
+  );
+}
+
 Piezo.prototype.tone = function(tone, duration) {
-  if (isNaN(tone) || isNaN(duration)) {
-    // Very Bad Things happen if one tries to play a NaN tone
-    throw new Error(
-      "Piezo.tone: invalid tone or duration"
-    );
+  if (this.frequency === Piezo.prototype.frequency) {
+    throwNeedsToneFrequencyError();
   }
 
-  clearTimer.call(this);
-
-  var timer = this.timer = new Timer();
-  var value = 1;
-
-  timer.setInterval(function() {
-    value = value === 1 ? 0 : 1;
-    this.io.digitalWrite(this.pin, value);
+  return this.frequency(toneToFrequency(tone), duration);
+};
 
-    if ((timer.difTime / 1000000) > duration) {
-      clearTimer.call(this);
-    }
-  }.bind(this), null, tone + "u", function() {});
+Piezo.prototype.frequency = function(frequency, duration) {
```

Comment:
One quirky part of this implementation I'd like thoughts on / alternatives, this requires the given controller to define at least one of `frequency` and `tone`. The Circuit Playground's sysex API, for example, works with `Piezo.prototype.frequency`-equivalent values (called `tone` in [Arduino land](https://www.arduino.cc/en/Reference/Tone)), while the internals of the default/digital piezo implementation work at the `Piezo.prototype.tone` level and treat frequency as the helper.

The intent with this approach is to reconcile controller implementations wanting to work with either tone OR frequency, without implementations needing to re-define or import the conversion methods.

Alternatively we could treat `Piezo.prototype.tone` as the only controller override-able method, just not sure whether that's more likely to be cumbersome for future controllers than `Piezo.prototype.frequency`.